Rename all instances of binormal as bitangent

shehzan10 opened the pull request:
Fixes #4822 

The PR deprecates `GeometryPipeline.createBinormalAndTangent` and recommends using the new `GeometryPipeline.createTangentAndBitangent` function.

The attributes have also been renamed to bitangent. This could cause some breakage.

This PR also makes the default value for `DebugAppearance.perInstanceAttribute` to `false` (was previously a required attribute).

Note: This PR fixes trailing whitespace in the files modified.

pjcozzi commented:
@austinEng can you do the first review on this?  Then I will do a final review and merge.

austinEng commented on Source/Core/EllipseGeometry.js:
```diff
@@ -175,15 +175,15 @@ define([
-                            binormals[i + bottomOffset] = binormal.x;
-                            binormals[i1 + bottomOffset] = binormal.y;
-                            binormals[i2 + bottomOffset] = binormal.z;
+                            bitangents[i  + bottomOffset] = bitangent.x;
```
Whitespace. This occurs several other times, but other than that, I think this looks good.

shehzan10 replied:
I've actually done that intentionally to have the code better aligned.

mramato replied:
I appreciate the eye for code formatting and cleanliness, but as a general rule we try to rely on auto-formatting (Webstorm CTRL-SHIFt-L) for blocks of code because it makes it more consistent across the code base as a whole. Unless we starting formatting array lookups like this everywhere, it's not ideal to only do it in a few places (or only have certain devs do it). A good mantra we try to follow is that the code should look like it was all written by one person, even though over 100 people have contributed so far.

That being said, in the future I'm hoping we can move to eslint and you that to enforce common style throughout the codebase.  These types of rules would then become trivial to add or enforce.

pjcozzi commented on CHANGES.md:
```diff
@@ -4,8 +4,10 @@ Change Log
+    * The function `createBinormalAndBitangent` of `GeometryPipeline` will be removed in 1.31. Use the function `createTangentAndBitangent` instead. [#4856](https://github.com/AnalyticalGraphicsInc/cesium/pull/4856)
 * Breaking changes
     * Removed separate `heading`, `pitch`, `roll` parameters from `Transform.headingPitchRollToFixedFrame` and `Transform.headingPitchRollQuaternion`. Pass a `headingPitchRoll` object instead. [#4843](https://github.com/AnalyticalGraphicsInc/cesium/pull/4843)
+    * The property `binornmal` has been renamed to `bitangent` throughout. [#4856](https://github.com/AnalyticalGraphicsInc/cesium/pull/4856)
```
Be more precise. `Geometry`, `VertexFormat`, etc.  Otherwise, it isn't obvious to users where this occurs.

pjcozzi commented:
Please submit an issue to fully remove the deprecated code.  For an example, see #4730.

pjcozzi commented:
In addition to the unit tests, do a bit of manual testing here.  Open some of the geometry examples in Sandcastle, make sure to use a `VertexFormat` that includes tangent-space vectors, then visualize them with [DebugAppearance](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Scene/DebugAppearance.js) and verify this branch is the same as master.

pjcozzi commented on Specs/Core/GeometryPipelineSpec.js:
```diff
@@ -1621,18 +1621,57 @@ defineSuite([
                                                         -0.4082482904638631, -0.8164965809277261, 0.4082482904638631], CesiumMath.EPSILON7);
     });
 
+    it ('computeTangentAndBitangent computes tangent and bitangent for an BoxGeometry', function() {
```
Can you fix this typo throughout the old test code: "an BoxGeometry'" -> "a BoxGeometry'"

pjcozzi commented:
Just those comments.  Nice job!

shehzan10 commented:
I have changed `DebugAppearance.perInstanceAttribute` to be `false` by default based on @pjcozzi's suggestion. Added this to the changelog as well.

@mramato Thanks for that heads up. I have reverted those spaces in this PR.

pjcozzi commented:
Looks good.  I also noticed this test failure in master, could you please open a separate PR to fix it?

```
Scene/EllipsoidPrimitive Constructs with options
DeveloperError: Expected scale to be typeof object, got number
Error
    at new DeveloperError (http://localhost:8080/Source/Core/DeveloperError.js:44:19)
    at Object.Check.typeOf.object (http://localhost:8080/Source/Core/Check.js:128:19)
    at Function.Matrix4.fromScale (http://localhost:8080/Source/Core/Matrix4.js:453:22)
    at http://localhost:8080/Specs/Scene/EllipsoidPrimitiveSpec.js:58:35
    at Object.<anonymous> (http://localhost:8080/Specs/spec-main.js:139:30)
    at attemptAsync (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:1793:24)
    at QueueRunner.run (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:1746:26)
    at QueueRunner.execute (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:1732:10)
    at Spec.queueRunnerFactory (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:620:35)
    at Spec.execute (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:357:10)
```
